plasma_cash/client/history.py

```diff
@@ -0,0 +1,37 @@
+import rlp
+from rlp.sedes import big_endian_int, binary, CountableList
+
+from .exceptions import TxHistoryNotFoundException
+
+
+class History(rlp.Serializable):
+    EMPTY_TX = b'\x00' * 32
+
+    fields = [
+        ('deposit_tx_blk_num', big_endian_int),
+        ('latest_tx_blk_num', big_endian_int),
+        ('latest_blk_num', big_endian_int),
+        ('tx_history', CountableList(binary)),
+        ('proofs', CountableList(binary))
+    ]
+
+    def __init__(self, deposit_tx_blk_num, tx, proof):
+        self.deposit_tx_blk_num = deposit_tx_blk_num
+        self.latest_tx_blk_num = deposit_tx_blk_num
+        self.latest_blk_num = deposit_tx_blk_num
+        self.tx_history = [tx]
+        self.proofs = [proof]
+
+    def update_tx_history(self, tx, proof):
+        # TODO: should add re-org protection
```

Comment:
I don't think there's too much concern of re-org here. The block number using here is Plasma chain's block num. Even root chain re-orgs as long as the block is still included in root chain it would be fine. The only concern is that after re-org the history here is not on the chain anymore, which might not be something we need to worry too much?